extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSinkFactory.java

```diff
@@ -99,22 +102,25 @@ public DataSink createSink(DataFlowStartMessage request) {
         return validation.validate(destination).flatMap(ValidationResult::toResult);
     }
 
-    private S3Client createS3Client(DataAddress destination) {
-
-        String endpointOverride = destination.getStringProperty(ENDPOINT_OVERRIDE);
-
-        S3Client client;
-        var secret = vault.resolveSecret(destination.getKeyName());
-        if (secret != null) {
-            var secretToken = typeManager.readValue(secret, AwsTemporarySecretToken.class);
-            client = clientProvider.s3Client(S3ClientRequest.from(destination.getStringProperty(REGION), endpointOverride, secretToken));
-        } else if (credentialsValidation.validate(destination).succeeded()) {
-            var secretToken = new AwsSecretToken(destination.getStringProperty(ACCESS_KEY_ID),
-                    destination.getStringProperty(SECRET_ACCESS_KEY));
-            client = clientProvider.s3Client(S3ClientRequest.from(destination.getStringProperty(REGION), endpointOverride, secretToken));
+    private S3Client createS3Client(DataAddress address) {
```

Comment:
for sake of simplicity, this method could return a `S3ClientRequest` and let the caller call the `clientProvider`